addons/purchase_mrp/tests/test_anglo_saxon_valuation.py

```diff
@@ -273,3 +273,274 @@ def test_fifo_cost_adjust_mo_quantity(self):
         manufacturing_order.move_raw_ids.quantity = 1
 
         self.assertEqual(self.product_a.standard_price, 100)
+
+    def test_purchase_kit_cumulative_cost_share_with_auto_avco_components_bom_repetition(self):
+        """
+        A kit Super Kit with three AVCO components
+        - C01, cost share 10%
+        - A kit Sub Kit 1 with three AVCO components, cost share 30%
+            - C02, cost share 50%
+            - C03, cost share 25%
+            - Sub Kit 2 with two AVCO components, cost share 25%
+                - C04, cost share 30%
+                - C05, cost share 70%
+        - Sub Kit 2 with two AVCO components, cost share 60%
+                - C04, cost share 30%
+                - C05, cost share 70%
+        Buy and receive 1 kit Super Kit @ 1000
+        """
+        component01, component02, component03, component04, component05 = self.env['product.product'].create([{
+            'name': 'Component %s' % name,
+            'categ_id': self.avco_category.id,
+        } for name in ['01 (super kit)', '02 (sub kit 1)', '03 (sub kit 1)', '04 (sub kit 2)', '05 (sub kit 2)']])
+
+        super_kit, sub_kit_1, sub_kit_2 = self.env['product.product'].create([{
+            'name': name,
+        } for name in ['Super Kit', 'Sub Kit 1', 'Sub Kit 2']])
+
+        self.env['mrp.bom'].create({
+            'product_tmpl_id': super_kit.product_tmpl_id.id,
+            'type': 'phantom',
+            'bom_line_ids': [Command.create({
+                'product_id': component01.id,
+                'product_qty': 1,
+                'cost_share': 10,
+            }), Command.create({
+                'product_id': sub_kit_1.id,
+                'product_qty': 1,
+                'cost_share': 30,
+            }), Command.create({
+                'product_id': sub_kit_2.id,
+                'product_qty': 1,
+                'cost_share': 60,
+            })],
```

Comment:
Nitpick but imo this is nicer and more clear to read as such:
```suggestion
            'bom_line_ids': [
                Command.create({'product_id': component01.id, 'product_qty': 1, 'cost_share': 10}),
                Command.create({'product_id': sub_kit_1.id, 'product_qty': 1, 'cost_share': 30}),
                Command.create({'product_id': sub_kit_2.id, 'product_qty': 1, 'cost_share': 60}),
            ],
```
The same holds for other occurrences of Command in this file.